homeassistant/components/konnected/panel.py

```diff
@@ -0,0 +1,365 @@
+    CONF_MOMENTARY,
+    CONF_PAUSE,
+    CONF_POLL_INTERVAL,
+    CONF_REPEAT,
+    DOMAIN,
+    ENDPOINT_ROOT,
+    SIGNAL_SENSOR_UPDATE,
+    STATE_LOW,
+    ZONE_TO_PIN,
+)
+from .errors import CannotConnect
+
+_LOGGER = logging.getLogger(__name__)
+
+KONN_MODEL = "Konnected"
+KONN_MODEL_PRO = "Konnected Pro"
+
+# Indicate how each unit is controlled (pin or zone)
+KONN_API_VERSIONS = {
+    KONN_MODEL: CONF_PIN,
+    KONN_MODEL_PRO: CONF_ZONE,
+}
+
+
+class AlarmPanel:
+    """A representation of a Konnected alarm panel."""
+
+    def __init__(self, hass, config_entry):
+        """Initialize the Konnected device."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.config = (
+            config_entry.data
+        )  # a configuration.yaml device config contained in a device entry
+        self.host = self.config.get(CONF_HOST)
+        self.port = self.config.get(CONF_PORT)
+        self.client = None
+        self.status = None
+        self.api_version = KONN_API_VERSIONS[KONN_MODEL]
+
+    @property
+    def device_id(self):
+        """Device id is the MAC address as string with punctuation removed."""
+        return self.config.get(CONF_ID)
+
+    @property
+    def stored_configuration(self):
+        """Return the configuration stored in `hass.data` for this device."""
+        return self.hass.data[DOMAIN][CONF_DEVICES].get(self.device_id)
+
+    def format_zone(self, zone, other_items=None):
+        """Get zone or pin based dict based on the client type."""
+        payload = {
+            self.api_version: zone
+            if self.api_version == CONF_ZONE
+            else ZONE_TO_PIN[zone]
+        }
+        payload.update(other_items or {})
+        return payload
+
+    async def async_connect(self):
+        """Connect to and setup a Konnected device."""
+        try:
+            import konnected
+
+            self.client = konnected.Client(
+                host=self.host,
+                port=str(self.port),
+                websession=aiohttp_client.async_get_clientsession(self.hass),
+            )
+            self.status = await self.client.get_status()
+            self.api_version = KONN_API_VERSIONS.get(
+                self.status.get("model", KONN_MODEL), KONN_API_VERSIONS[KONN_MODEL]
+            )
+            _LOGGER.info(
+                "Connected to new %s device", self.status.get("model", "Konnected")
+            )
+            _LOGGER.info(self.status)
+
+            await self.async_update_initial_states()
+            # brief delay to allow processing of recent status req
+            await asyncio.sleep(0.1)
+            await self.async_sync_device_config()
+
+        except self.client.ClientError as err:
+            _LOGGER.warning("Exception trying to connect to panel: %s", err)
+            raise CannotConnect
+
+        _LOGGER.info(
+            "Set up Konnected device %s. Open http://%s:%s in a "
+            "web browser to view device status.",
```

Comment:
We don't end logging messages with period.

Reply:
👍